### ISSUE TITLE
BUG: Results window content going off screen

### ISSUE THREAD

**Issue by dsanalytics** [DonJayamanne/vscodeJupyter#88]
BUG: Results window content going off screen!

If Append Results option is checked, Result window does not scroll to the end, so new results eventually go off screen and manual scrolling is required - this is yet another thing that makes VSCode UNUSABLE for me at this point. You may ask, why don’t you just uncheck Append Results, to which I can report that then I get hit by another bug in your extension described here (https://github.com/DonJayamanne/vscodeJupyter/issues/82). So these two bugs in their handy combination make VSCode UNUSABLE for my work, thus I'm currently forced to use Atom/Hydrogen for they work properly. 

So your prompt action would be appreciated on both of these issue for they are unavoidable deal-breakers!

Environment:
Visual Studio Code v1.18.0
Jupyter Extension v1.1.4
Python v3.6.2
matplotlib v2.1.0
OS 1: Windows 7 SP1 64bit Home Premium
OS 2: Windows 10 64bit Home Premium
Jupyter Notebook Server v5.0.0

P.S. Congrats on your new employment with MS Don! Hopefully, now we won't be forced to wait weeks and months for one-liner fixes, while Atom/Hydrogen are eating VSCode lunch.

![vsc-bug-results-off-screen-annotated](https://user-images.githubusercontent.com/32625394/32903849-b2f8a26a-cabb-11e7-8a77-e0764ccedff2.png)

**Comment by dsanalytics** [comment]
Don, would you mind fixing this at your earliest convenience? It's a tremendous time waster to all of us having to scroll every single time to see results. I'm sure it's just one line - auto scroll to the end after code execution finishes.

P.S. While you at it, perhaps fix the other one-liner-fix bug https://github.com/DonJayamanne/vscodeJupyter/issues/82 ?

Thanks in advance.

**Comment by dakl** [comment]
I love this plugin but this is driving me absolutely bananas. I would love to see a fix. 

Thanks!

**Comment by anongit** [comment]
One-liner fix would be to prepend results instead of append.

**Comment by dsanalytics** [comment]
@anongit VSCode should behave the way Atom editor is handling it and most other serious editors I'm sure - just auto-scroll to the end after appending results and that's one-liner.

@DonJayamanne - I just don't get it - can you please fix this and https://github.com/DonJayamanne/vscodeJupyter/issues/82 too?

**Comment by anongit** [comment]
@dsanalytics Fair enough. Auto-scroll to bottom is already merged in #97 as it turns out.

**Comment by dsanalytics** [comment]
Thanks @anongit I have not seen any update notifications in VSCode for Jupyter in a while - so this has not been released officially. I'm not sure why the silence and no response from the author at all. And he's working for MS now if you'd believe it. What is going on?

Plus #82 still remains as a bug and it's been months.

P.S. This is pretty sad - many of these markets are completely open for the taking (especially Data Science) while MS is busy fixing Javascript (presumably to displace C#/ASP.NET) and making apps for Android. Windows Phone has already been brought down - presumably to allow competitors to thrive. They must be glad that most shareholders are not tech-savvy and don't see this leadership disaster unravel following BG departure.

**Comment by dakl** [comment]
Should we consider this abandonware then?

**Comment by dsanalytics** [comment]
@dakl - not sure. Currently, my primary DS tool is Atom editor + Hydrogen (atom extension equivalent of VSCode Jupyter) + Pweave (atom extension I use for DS reporting). As you may know, Atom and VSCode share the same base (Electron). So check it out if your primary work is Data Science (EDA, algo design, etc).

**Comment by Apollo3zehn** [comment]
@dsanalytics - as a workaround, you can add the fix manually:

- Open %userprofile%\.vscode\extensions\donjayamanne.jupyter-1.1.4\out\browser\bundle.js in an editor.
- search for the string `n.props.resultActions.addResults(e)` and append the following string: `,window.scrollTo(0, document.body.scrollHeight)`
- restart VS Code

Basically I did the same as done here: https://github.com/DonJayamanne/vscodeJupyter/commit/508ca40c862f37b0182d55108534ea688d5d4173

**Comment by dsanalytics** [comment]
Thanks @Apollo3zehn - it worked!

But this is still not there yet compared to Atom. Also, it's been a while since last Jupyter update, and investing time in a tool/add-on that seems to be abandonware is low ROI. I may stick around for python debugging as Atom may be lacking there. It's just funny how MS tends to please competitors (OSS, Javascript, Android) while abandoning people that use MS tech - see my previous post here.

**Comment by neilsustc** [comment]
FYI, [PR#97 comment](https://github.com/DonJayamanne/vscodeJupyter/pull/97#issuecomment-425156930).
